### ISSUE TITLE
[Bug]: Numerous Broken Fragment IDs Within Documentation

### ISSUE THREAD

**Issue by jjanzenn** [puppetlabs/forge_issues#32]
### What happened?

I have noticed a very large number of incorrect fragment IDs in links under the reference tab of documentation. For instance, in the documentation for [stdlib](https://forge.puppet.com/modules/puppetlabs/stdlib/reference), there is a link to the section on `stdlib::stages`. The link does not take you anywhere though because it uses the fragment ID `#stdlib--stages` in its URL. There is the fragment ID `#stdlibstages` in the page which is where the URL should be pointing. Most of the links are correct on that module, but I have come across this exact issue within the documentation for many projects (for instance, see almost all the links in <https://forge.puppet.com/modules/treydock/slurm/reference>). It seems that when generating the reference page from `REFERENCE.md`, the links are created incorrectly.

### What browsers / clients are you seeing the problem on?

Firefox

**Comment by skarwaperforce** [comment]
@JacobJanzen I agree, there seems to be an issue. Many thanks for highlighting this, really appreciate this. I would mark this as Triaged and we can pick this up in one of our next iterations. Thanks again!
